src/ape/api/providers.py

```diff
@@ -1316,6 +1386,109 @@ def _find_txn_by_account_and_nonce(
                     stop_block,
                 )
 
+    def poll_blocks(
+        self,
+        stop_block: Optional[int] = None,
+        required_confirmations: Optional[int] = None,
+        new_block_timeout: Optional[int] = None,
+    ) -> Iterator[BlockAPI]:
+        network_name = self.network.name
+        block_time = self.network.block_time
+        wait_time = block_time / 2
+        timeout = (
+            (
+                10.0
+                if network_name == LOCAL_NETWORK_NAME or network_name.endswith("-fork")
```

Comment:
**note for future work or myself**: At this point, we should have an `is_local` property on `NetworkAPI`.

Reply:
https://github.com/ApeWorX/ape/pull/1730